test/general/macros.pass.cpp

```diff
@@ -0,0 +1,56 @@
+// -*- C++ -*-
+//===-- macros.pass.cpp --------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+
+#include <oneapi/dpl/pstl/onedpl_config.h>
+
+#include "support/utils.h"
+
+#if  !_ONEDPL_PAR_BACKEND_SERIAL
+    #ifdef _ONEDPL_PARALLEL_BACKEND_SERIAL_H
+        #error The parallel serial backend is used while it should not (_ONEDPL_PAR_BACKEND_SERIAL==0)
+    #endif
+#endif
+
+#if  !_ONEDPL_PAR_BACKEND_OPENMP

```

Comment:
These checks are useful, thank you @KseniiaBakina!

In addition, It may be useful to verify if we enable a proper backend for parallel policies depending on the use of ```ONEDPL_USE_TBB_BACKEND``` and  ```ONEDPL_USE_OPENMP_BACKEND``` macros following the table

| OpenMP backend / TBB backend         | 0                                             | Undefined   | 1          |
|:------------------------------------:|:--------------------------------------:|:-----------:|:----------:|
|0                                     |Serial                                                               | TBB           | TBB         |
|Undefined                      |OpenMP if available.</br> Otherwise, serial | TBB            | TBB         |
|1                                     |OpenMP                                                         | OpenMP    | TBB         |

from #414

Reply:
You are right, such a check will be useful, but it is worth creating a separate test for it. I guess we shouldn't include it in this patch.

Reply:
Feel free to do this in a separate patch. As for creation of a separate test, I think it is not necessary, because all the checks are macros based and the proposed ones extend checks from this patch.

Reply:
Decided to check public API basing on explicit definition of macros